### PR TITLE
Adding notes from two sessions

### PR DESCRIPTION
**Pull request by speedoholic**
- Explore App Clips
- Meet the new photos picker

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+- **New App Clip Target in Xcode**
+    - Contains all assets
+    - Needs to be submitted along with the app for review
+    - It gets downloaded separately if the app is not installed on user’s device
+    - Should be as small as possible for quick downloads (less than 10MB but with enough assets to load UI quickly)
+    - Focused user flows - one at a time
+- **Demo**
+    - Ordering smoothies
+    - New App Clip target embedded in the application
+        - Name and bundle ID for app clip added
```
Comment (rogerluan) [inline README.md]:
```suggestion
        - Name and bundle ID for App Clip added
```

File: README.md [hunk 2]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - It can build and run with boilerplate code right away
+    - Add code and resources
+        - `NutritionFacts` dependency added
+        - Create new Assets Catalog as shared assets
+            - Drag App Icon, Colors and other image assets required for app clip into the shared assets catalog
+        - Add the required model and view files to App Clip target
+            - Let go of unwanted swift files like navigation
+        - Conditionally compile out the references to files not added to App Clip
+            - Build Settings > Swift Compiler custom flags > Active Compilation Conditions > Add APPCLIP condition for required schemes.
+            - Use #if !APPCLIP to compile out unwanted references in swift code
```
Comment (rogerluan) [inline README.md]:
```suggestion
            - Use #if !APPCLIP to compile out unwanted references in Swift code
```

File: README.md [hunk 3]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - Write code for AppClip
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - Write code for App Clip
```

File: README.md [hunk 4]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - Add required models and views to the new AppClip
+        - Include existing views in the AppClip’s content view
```
Comment (rogerluan) [inline README.md]:
```suggestion
        - Include existing views in the App Clip’s content view
```

File: README.md [hunk 5]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+- **Technology Overview**
+    - AppClips are built using same UI components as an App
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - App Clips are built using same UI components as an app
```

File: README.md [hunk 6]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - When launched, it receives `NSUserActivity`
+        - Use the URL to identify the type of experience to be handled
+    - Unlike extensions, AppClip can make use of all iOS SDK APIs
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - Unlike extensions, App Clip can make use of all iOS SDK APIs
```

File: README.md [hunk 7]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - Note: Access to sensitive data is limited
+        - Always check if data is available
+    - New location confirmation API helps get the location quickly without requesting full access
+    - New API for migrating data from AppClip to main app using shared data container once installed
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - New API for migrating data from App Clip to main app using shared data container once installed
```

File: README.md [hunk 8]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - AppClip and it’s local storage will be deleted after period of inactivity
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - App Clip and its local storage will be deleted after period of inactivity
```

File: README.md [hunk 9]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - Not included in backups
+    - Can not be launched via Universal links or custom URL schemes
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - Can not be launched via Universal Links or custom URL schemes
```

File: README.md [hunk 10]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - Can not include bundle extensions like content blockers
+- **Device states and transitions**
+    - User scans QR code
+    - iOS locates, downloads and runs the app clip
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - iOS locates, downloads and runs the App Clip
```

File: README.md [hunk 11]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - If the app clip is not revisited for a while, the app clip and it’s data gets deleted
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - If the App Clip is not revisited for a while, the App Clip and its data gets deleted
```

File: README.md [hunk 12]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - Treat app clip data as cache; which can be deleted
```
Comment (rogerluan) [inline README.md]:
```suggestion
        - Treat App Clip data as cache; which can be deleted
```

File: README.md [hunk 13]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - If the app clip is visited frequently, the app clip lifetime will be extended and data may never be cleared
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - If the App Clip is visited frequently, its lifespan will be extended and data may never be cleared
```

File: README.md [hunk 14]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - When user downloads your app, iOS will automatically migrate the data container and the user permissions granted by user already
```
Comment (rogerluan) [inline README.md]:
```suggestion
        - When user downloads your app, iOS will automatically migrate the data container and the permissions that were already granted by the user
```

File: README.md [hunk 15]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - iOS still deletes the app clip and it’s container after copying the data container to the app
```
Comment (rogerluan) [inline README.md]:
```suggestion
        - iOS still deletes the App Clip and its container after copying the data container to the app
```

File: README.md [hunk 16]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+- **Other technologies**
+    - Apple Pay
+    - Notifications
+    - SwiftUI
+    - `SKOverlay` / AppStoreOverlayModifier (Refer to what’s new in In-app purchase session)
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - `SKOverlay` / `AppStoreOverlayModifier` (refer to [What’s new with in-app purchase](https://developer.apple.com/wwdc20/10661/) session)
```

File: README.md [hunk 17]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - `ASAuthorizationController` for sign in or sign up
+
+
+## Meet the new Photos Picker
+
+https://developer.apple.com/videos/play/wwdc2020/10652/
```
Comment (rogerluan) [inline README.md]: Please refer to the [Contributing.md](https://github.com/Blackjacx/WWDC/blob/master/CONTRIBUTING.md#submitting-changes) file 😃 
```suggestion
https://developer.apple.com/wwdc20/10652/
```

File: README.md [hunk 18]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+        - iOS still deletes the app clip and it’s container after copying the data container to the app
+- **Other technologies**
+    - Apple Pay
+    - Notifications
+    - SwiftUI
+    - `SKOverlay` / AppStoreOverlayModifier (Refer to what’s new in In-app purchase session)
+    - `ASAuthorizationController` for sign in or sign up
+
+
+## Meet the new Photos Picker
```
Comment (rogerluan) [inline README.md]:
```suggestion
## Meet the new Photos picker
```

☝️ consistent with the WWDC website

File: README.md [hunk 19]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+
+- **PHPicker** 
+  - Direct access to user’s photo gallery
+	- Supports zoom in, multi-select, review
+	- Types are filterable
+	- Privacy built in by default
+	- It won’t prompt the user for access
+	- It runs out of the app’s process
+		- Separate process rendered on top of the app
+		- Only what user selects is passed back to the app
```
Comment (rogerluan) [inline README.md:639]: This is awesome!! 🤩 finally!

File: README.md [hunk 20]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+- **Implementation**
+  - `PHPickerConfiguration` which can include type filters is passed to `PHPickerViewController` which has delegates to handle responses
```
Comment (rogerluan) [inline README.md]:
```suggestion
  - `PHPickerConfiguration` (which can include type filters) is passed to `PHPickerViewController`, which has delegates to handle responses
```

File: README.md [hunk 21]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+		- Initialize the `PHPickerViewController` with this configuration
+		- Present the new controller after setting delegate and following the protocol
+	  - Implement `didFinishPicking`
+			- Dismiss the picker first
+			- Retrieve the image via item provider and use main queue to update your UI
+	  - Use `IndexingIterator<[NSItemProvider]>` to save array of item providers
+	  - Add touch event to iterate through the array of picked images
+- **Note**
+  - `AssetsLibrary` will go away; switch to `PhotoKit`
+  - `UIImagePickerController` is to be deprecated and replaced with `PHPickerViewController`
```
Comment (rogerluan) [inline README.md]: Great! This is good to know 🚀

File: README.md [hunk 22]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+
+_James Savage, Luming Yin_
+
+- **What is an App Clip?**
+    - App Clip Experience URL is required to run App Clips which are parts of your App
+    - On-demand app experiences
+- **App Clip Experiences**
+    - App Clip URLs are similar to Universal Links
+    - Registered using App Store connect
+    - Surfaced through user actions via NFC / QR codes / Links in safari or other apps / Apple App Clip codes combine the ease of NFC and visual codes so they can be tapped or scanned
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - Surfaced through user actions via NFC / QR codes / links in safari or other apps / Apple App Clip codes combine the ease of NFC and visual codes so they can be tapped or scanned
```

File: README.md [hunk 23]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+
+
+## Explore App Clips
+
+https://developer.apple.com/wwdc20/10174
+
+_James Savage, Luming Yin_
+
+- **What is an App Clip?**
+    - App Clip Experience URL is required to run App Clips which are parts of your App
```
Comment (rogerluan) [inline README.md]:
```suggestion
    - App Clip Experience URL is required to run App Clips which are parts of your app
```

**Review by rogerluan — CHANGES_REQUESTED** [review]
Explore app clips was next on my list, glad you beat me to it (and I saw your PR before starting 😂 )

Two notes about my review:
- Although they look like too many comments, they're mostly just styling comments. You can apply all my suggestions with a single button directly from GitHub website 👍 
- The content is great! 🚀 

Thanks for the submission 💪

**Comment by Blackjacx** [comment]
> Although they look like too many comments, they're mostly just styling comments. You can apply all my suggestions with a single button directly from GitHub website 👍

@rogerluan I applied all your changes one by one. Now I understand why you suggested the spell check editor 👍 MAy I ask which one you prefer?

> Explore app clips was next on my list, glad you beat me to it (and I saw your PR before starting 😂 )

I added a hint to the CONTRIBUTING.md to look at the open PR's if a session has already been processed.

File: README.md [hunk 24]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+https://developer.apple.com/wwdc20/10174
+
+_James Savage, Luming Yin_
+
+- **What is an App Clip?**
+    - App Clip Experience URL is required to run App Clips which are parts of your app
+    - On-demand app experiences
+- **App Clip Experiences**
+    - App Clip URLs are similar to Universal Links
+    - Registered using App Store connect
```
Comment (Blackjacx) [inline README.md]:
```suggestion
    - Registered using App Store Connect
```

File: README.md [hunk 25]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - Surfaced through user actions via NFC / QR codes / links in safari or other apps / Apple App Clip codes combine the ease of NFC and visual codes so they can be tapped or scanned
```
Comment (Blackjacx) [inline README.md]:
```suggestion
    - Surfaced through user actions via NFC / QR codes / links in Safari or other apps / Apple App Clip codes combine the ease of NFC and visual codes so they can be tapped or scanned
```

File: README.md [hunk 26]
```diff
@@ -551,4 +554,111 @@ _Jonathan Hersh, Allen Langmaier_
...
+    - Ordering smoothies
+    - New App Clip target embedded in the application
+        - Name and bundle ID for App Clip added
+        - It can build and run with boilerplate code right away
+    - Add code and resources
+        - `NutritionFacts` dependency added
+        - Create new Assets Catalog as shared assets
+            - Drag App Icon, Colors and other image assets required for app clip into the shared assets catalog
+        - Add the required model and view files to App Clip target
+            - Let go of unwanted swift files like navigation
```
Comment (Blackjacx) [inline README.md]:
```suggestion
            - Let go of unwanted Swift files like navigation
```

**Review by Blackjacx — CHANGES_REQUESTED** [review]
As @rogerluan said, really great content 👍 I just found some minor typos. Thank you so much 🙏

**Comment by rogerluan** [comment]
> @rogerluan I applied all your changes one by one. 

Oh boy 😱 when the "git diff" suggestions are still there, you can click on "Files" (top of this page), and click the "Add suggestion to batch" (or something like this). Do this to every suggestion you wanna apply, then hit "apply" on the top of the page so you have 1 commit with all the suggestions applied at once :) it's a lot faster specially because applying the commits 1 by one requires N full page loads hehe. I hope this helps next time 😉 

>Now I understand why you suggested the spell check editor 👍 MAy I ask which one you prefer?

I use https://atom.io with an add-on to format markdown docs 👍 it has built-in spell check already, but the markdown preview is a 3rd party plugin.

> I added a hint to the CONTRIBUTING.md to look at the open PR's if a session has already been processed.

Oh this is great! Nice addition! 🚀

**Comment by speedoholic** [comment]
Thanks @rogerluan and @Blackjacx for your appreciation and help on this.

**Pull request merged**